pre_commit/commands/run.py

```diff
@@ -104,9 +106,19 @@ def _run_single_hook(hook, repo, args, write, skips=frozenset()):
 
     write(color.format_color(pass_fail, print_color, args.color) + '\n')
 
-    if (stdout or stderr) and (retcode or args.verbose):
+    if (stdout or stderr or file_modifications) and (retcode or args.verbose):
         write('hookid: {0}\n'.format(hook['id']))
         write('\n')
+
+        # Print a message if failing due to file modifications
+        if file_modifications:
+            extra_text = (
+                ' Additional output:\n' if (stdout or stderr) else ''
+            )
+
+            write('Files were modified by this hook.{0}'.format(extra_text))
```

Comment:
instead of formatting, I'd just do

```python
if stdout or stderr:
    write(' Additional output:')
```

This'll avoid the slightly-awkward double linebreak in this case